src/test/utilities/collectionUtils.ts

```diff
@@ -3,6 +3,7 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
+// TODO: what is the point of this? should it live in src/shared/utilities/collectionUtils.ts ?
```

Comment:
It's because generator functions are special JS constructs which are not super easy to create. This should only be used in test code. Probably better to remove it and instead make it easier for consuming code to handle `Promise<T[]> | AsyncGenerator<T, T | void>` instead. Another time though.